backends/tofino/bf-p4c/phv/utils/utils.cpp

```diff
@@ -1312,7 +1312,7 @@ std::optional<le_bitrange> PHV::AlignedCluster::validContainerStartRange(
     LOG5("\tField slice " << slice << " to be placed in container size " << container_size
                           << " slices has "
                           << (slice.validContainerRange() == ZeroToMax()
-                                  ? "no"
+                                  ? "no"_cs
```

Comment:
maybe we can fix the other way round here? so we won't need to call `cstring::to_string` on `slice.validContainerRange()`? What is the type of `slice.validContainerRange()`?

Reply:
I believe `cstring::to_string` will call `operator<<` to format this.

Reply:
I could implement something like 

```cpp
template <RangeUnit Unit, Endian Order>
inline std::string to_string(const HalfOpenRange<Unit, Order> &range) {
    std::ostringstream ss;
    ss << range;
    return ss.str();
}

template <RangeUnit Unit, Endian Order>
inline std::string to_string(const ClosedRange<Unit, Order> &range) {
    std::ostringstream ss;
    ss << range;
    return ss.str();
}
```
in lib/bitrange.h to enable to_string methods.

Reply:
Let's do this in a separate PR. I will merge this one to fix the build.